src/Battlescape/TileEngine.cpp

```diff
@@ -4106,6 +4115,27 @@ bool TileEngine::medikitUse(BattleAction *action, BattleUnit *target, BattleMedi
 	
 	return canContinueHealing;
 }
+	
+void TileEngine::scriptedItemUse(BattleAction *action)
+{
+	BattleItem *item = action->weapon;
+	BattleActionType actionType = action->type;
+	bool canPerformAction = true;
+	ModScript::ScriptedItemUse::Output args { actionType, canPerformAction };
+	
+	ModScript::ScriptedItemUse::Worker work { action->actor, action->weapon, _save };
+	
+	work.execute(item->getRules()->getScript<ModScript::ScriptedItemUse>(), args);
+	
+	int newActionType = args.getFirst();
```

Comment:
This is where the "scriptable" part really comes into play. Even if the action the player selected was BA_USE, the new action returned by the script can be something entirely different.

That leaves the action definitions on the item with a default behavior (the BA that was defined) and a layout option. Firearm actions display an accuracy while the use action does not, for example.

Reply:
And how am I supposed to display this in Stats for Nerds, document in Ufopedia ruleset reference, check in the ruleset validator, describe on the forum, and support from now on until oxce death? It's basically unsupportable. Sigh.